bundler/lib/bundler/rubygems_ext.rb

```diff
@@ -266,23 +266,27 @@ def ===(other)
   end
 
   Platform.singleton_class.module_eval do
-    unless Platform.singleton_methods.include?(:match_spec?)
-      def match_spec?(spec)
-        match_gem?(spec.platform, spec.name)
-      end
+    if Platform.singleton_methods.include?(:match_spec?)
+      remove_method :match_spec?
+      remove_method :match_gem?
+      remove_method :match_platforms?
+    end
 
-      def match_gem?(platform, gem_name)
-        match_platforms?(platform, Gem.platforms)
-      end
+    def match_spec?(spec)
+      match_gem?(new(spec.platform), spec.name)
```

Comment:
Mmh, so `match_spec?` was always broken before that?
I think back then I thought `spec.platform` must return a Gem::Platform object and not a raw string (a bit surprising to say the least).

Reply:
Yeah, agreed. This only happens for `Bundler::EndpointSpecification` where the parsed platform from the API is not instantiated as a `Gem::Platform`. However, making that change caused issues and I'm not sure I'll have time to investigate those, so for now I went with this fix.

Reply:
https://github.com/rubygems/rubygems/blob/e5d979d31409a35443a24bf57ac9d1b438aa0f35/lib/rubygems/specification.rb#L2224-L2229 does seem to return a Gem::Platform, so probably the `spec` passed here is not a `Gem::Specification`.

It sounds like this should be handled in the caller, otherwise it's duplicating the given Gem::Platform object if called with a `Gem::Specification`.

Reply:
As I said, the problematic class is `Bundler::EndpointSpecification`. Also as I said, handling in the caller would be ideal but it caused some issues, so I went with this quick fix for now.

Reply:
Could we replace the call site(s) to `match_spec?`  which use a Bundler::EndpointSpecification to call `match_gem?(Gem::Platform.new(spec.platform), spec.name)` instead?
It feels a bit messy to monkey-patch RubyGems methods based on the Bundler version.

OTOH the current fix is more general and `Gem::Platform.new(aGemPlatform)` seems it should work besides creating a copy.

Reply:
I'll revisit this before the next release to try do something cleaner. If I can't figure it out we can go with this for now.

Reply:
Ok, I think I figured it out. Basically the specs that started failing after adding the fix to `Bundler::EndpointSpecification` were passing just by luck. It should be ok now.